Updated some missing params and return sections in the documentation.

nomad-software opened the pull request:
Added some missing parameters and return sections to the documentation.

In `std.conv` i've moved part of the doc block and the unittest to better document the `to` function within the `castFrom` template, a'la [`std.algorithm.aggregate`](http://dlang.org/phobos/std_algorithm_iteration.html#aggregate).

All changes can be previewed here: http://nomad.so/misc/d/dlang.org/fix-ddoc-parameter-warnings/phobos-prerelease/index.html

MetaLang commented on std/string.d:
```diff
@@ -1966,10 +1972,16 @@ auto representation(Char)(Char[] s) @safe pure nothrow @nogc
-    to lowercase.
- +/
+/**
+ * Capitalize the first character of $(D s) and convert the rest of $(D s)
```
This only works for characters in the ASCII range, right? If so, you should mention that.

nomad-software replied:
As far as I can see it supports unicode too.

MetaLang replied:
You're write, it does. I was on mobile and couldn't view the whole file. Never mind me.

MetaLang commented:
Doc LGTM minus capitalize

schuetzm commented on std/conv.d:
```diff
@@ -5275,15 +5275,18 @@ unittest
-                otherwise a compile-time error is emitted.
-
-    Returns:
-        the value after the cast, returned by reference if possible
```
Not sure about this. The template and its member are always to be used together. It could be confusing if the documentation is split. Have you tested how this appears in the generated documentation?

nomad-software replied:
This actually aids readability. Please read the initial pull description as there is a similar case and a link to a preview of the generate documentation.

schuetzm replied:
Ok, it's not ideal, but `to` was previously completely undocumented, it's definitely better now. LGTM then.

JakobOvrum commented on std/string.d:
```diff
@@ -1918,6 +1918,12 @@ ptrdiff_t lastIndexOfNeither(Char,Char2)(const(Char)[] haystack,
  * $(D ushort), or $(D uint) depending on the character width.
+ *
+ * Params:
+ *     s = The string to return the representation of.
```
s/representation/_representation/

JakobOvrum commented on std/string.d:
```diff
@@ -1918,6 +1918,12 @@ ptrdiff_t lastIndexOfNeither(Char,Char2)(const(Char)[] haystack,
+ *     s = The string to return the representation of.
+ *
+ * Returns:
+ *     The representation of the passed string.
```
ditto

JakobOvrum commented on std/string.d:
```diff
@@ -1966,10 +1972,16 @@ auto representation(Char)(Char[] s) @safe pure nothrow @nogc
+ * to lowercase.
+ *
+ * Params:
+ *     s = The string to capitalize.
```
s/capitalize/_capitalize/

JakobOvrum commented on std/bitmanip.d:
```diff
@@ -3191,6 +3191,7 @@ unittest
         T     = The integral type to convert the first $(D T.sizeof) bytes to.
         endianness = The endianness to write the bytes in.
         range = The range to write to.
+        value = The value to write.
```
s/value/_value/

JakobOvrum commented on std/bitmanip.d:
```diff
@@ -3536,6 +3537,7 @@ unittest
         T     = The integral type to convert the first $(D T.sizeof) bytes to.
         endianness = The endianness to write the bytes in.
         range = The range to append to.
+        value = The value to append.
```
s/append/_append/

The above existing line also makes the same mistake (this is such a misfeature of DDoc...)

nomad-software replied:
What does the underscore do?

JakobOvrum replied:
DDoc for a symbol `foo` highlights `foo` in the documentation description, assuming that every use of the word `foo` is a reference to the symbol itself. The same is done for parameter names. The underscore tells DDoc that the word is used in its original meaning, not as a reference to a symbol.

nomad-software replied:
Right, I get it, i'll amend it.

nomad-software replied:
Looking at the output, the parameter names are not highlighted in the same way as symbol names are so this might be a unnecessary change for parameter descriptions.

nomad-software commented:
Updated documentation to remove unnecessary symbol highlights in descriptions. I've updated the preview to, see the link in the original pull description.

JakobOvrum commented:
Auto-merge toggled on
